src/averageAges.js

```diff
@@ -15,11 +15,26 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter((p) => p.sex === 'm');
+
+  const menCentury = men.reduce((acc, m) => {
+    const centuryManDied
+      = m.died % 100 === 0 ? m.died / 100 : Math.floor(m.died / 100 + 1);
+
+    const isCorrectCentury = century ? centuryManDied === century : false;
+
+    return isCorrectCentury ? [...acc, m] : acc;
+  }, []);
```

Comment:
You can just use [Math.ceil](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Math/ceil) to avoid all these checks